src/charm.py

```diff
@@ -390,9 +395,12 @@ def set_signing_key(self, signing_key: str) -> None:
         if signing_key == self.get_signing_key():
             logger.info("Received signing key but there is no change, skipping")
             return
-        logger.debug("Adding signing key to secret: %s", signing_key)
-        secret = self.app.add_secret({"secret-signing-key": signing_key})
-        peer_relation[0].data[self.app].update({"secret-signing-id": typing.cast(str, secret.id)})
+        if self.unit.is_leader():
```

Comment:
I understand it is quite bad if we are here and the unit is not a leader

Reply:
Only leader can write data on the application peer relation data bag so it would fail anyway :|
The pitfall here is the main unit is not the leader and the secret is not there (kind of edge case but anyway, to be considered).

Reply:
I'll create a story for that, just in case to not forget about it.